frontend/lib/Views/AdminPage/GroupsView.dart

```diff
@@ -0,0 +1,159 @@
+
+class GroupsView extends StatefulWidget {
+  const GroupsView({required this.repository, Key? key}) : super(key: key);
+
+  final AdminGateway repository;
+
+  @override
+  State<GroupsView> createState() => _GroupsViewState();
+}
+
+class _GroupsViewState extends State<GroupsView> {
+  @override
+  Widget build(BuildContext context) => Scaffold(
+        appBar: _buildAppBar(),
+        body: _buildBody(),
+      );
+
+  PreferredSize _buildAppBar() => PreferredSize(
+        preferredSize: const Size.fromHeight(120),
+        child: AppBar(
+          centerTitle: true,
+          toolbarHeight: 120,
+          title: Container(
+              padding: const EdgeInsets.only(top: 20, bottom: 10),
+              child: const Text("Administrator panel")),
+          backgroundColor: Colors.white,
+          titleTextStyle: const TextStyle(
+              color: Colors.black,
+              fontFamily: 'Ubuntu Condensed',
+              fontSize: 40,
+              fontWeight: FontWeight.w500),
+          titleSpacing: 20,
+          bottom: PreferredSize(
+            preferredSize: const Size.fromHeight(20),
+            child: Container(
+              padding: const EdgeInsets.only(bottom: 10, top: 5),
+              child: const Text(
+                "User groups",
+                style: TextStyle(fontSize: 25),
+              ),
+            ),
+          ),
+        ),
+      );
+
+  FutureBuilder _buildBody() => FutureBuilder<List<GroupSummary>>(
+      future: widget.repository.getGroupsSummary(),
+      builder: (context, snapshot) {
+        if (snapshot.connectionState == ConnectionState.none ||
+            snapshot.data == null) {
+          return loadingInCenter();
+        }
+        return ChangeNotifierProvider(
+          create: (context) =>
+              GroupListProvider(snapshot.data!, widget.repository),
+          child: Consumer<GroupListProvider>(
+            builder: (context, groupListProvider, _) =>
+                _buildGroupList(groupListProvider, snapshot.data!.length),
+          ),
+        );
+      });
+
+  ListView _buildGroupList(
+          GroupListProvider groupListProvider, int itemCount) =>
+      ListView.builder(
+          itemCount: itemCount,
+          padding: const EdgeInsets.symmetric(horizontal: 20),
+          itemBuilder: (context, i) =>
+              _buildGroupCard(groupListProvider.groupsList[i]));
+
+  Card _buildGroupCard(GroupCard group) => Card(
+        margin: const EdgeInsets.symmetric(vertical: 10, horizontal: 10),
+        shadowColor: Colors.black,
+        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(10)),
+        child: ExpansionTile(
+          onExpansionChanged: (value) {
+            if (value) {
+              setState(() {
+                group.titleColor = Colors.pink;
+              });
+            } else {
+              setState(() {
+                group.titleColor = Colors.black;
+              });
+            }
```

Comment:
Można krócej :sweat_smile: 
```suggestion
setState(() {
  group.titleColor = value ? Colors.pink : Colors.black;
});
```